include/path_tracking_pid/visualization.hpp

```diff
@@ -0,0 +1,30 @@
+#include <string>
+
+#include "geometry_msgs/Pose.h"
+#include "ros/publisher.h"
+#include "std_msgs/ColorRGBA.h"
+#include "tf2/LinearMath/Transform.h"
+
+namespace path_tracking_pid
+{
+class Visualization
+{
+public:
+  Visualization(ros::NodeHandle nh);
+
+  void publishControlPoint(std_msgs::Header header, tf2::Transform pose);
+  void publishAxlePoint(std_msgs::Header header, tf2::Transform pose);
```

Comment:
Curious, why not send a TransformStamped?

Reply:
I think most of the internal datatypes should be `tf2::Transform` because you can perform calculations with them. I could also change this to `tf2::Stamped<tf2::Transform>` if you want

Reply:
Well, bonus of you current implementation is that the header only needs initialisation once in PID.